contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2882,6 +2882,9 @@ class tsqlMutator : public TSqlParserBaseListener
 				{
 					if (ctx->function_arg_list() && !ctx->function_arg_list()->expression().empty())
 					{
+						if (ctx->function_arg_list()->expression().size() != 3)
+							ereport(ERROR, (errcode(ERRCODE_UNDEFINED_FUNCTION), errmsg("The substring function requires 3 argument(s).")));
```

Comment:
why do we need to handle this specially here?

Reply:
Modified the handling to rewrite `substring` to `sys.substring`